doc/secret-keys.md

```diff
@@ -0,0 +1,91 @@
+
+For example:
+```yaml
+    roles:
+    - id: controller
+      members: 1
+      resources:
+        limits:
+          cpu: "1"
+          memory: 2Gi
+        requests:
+          cpu: "1"
+          memory: 2Gi
+      secretKeys:
+      - name: some-key-name
+        value: some-key-value
+      - name: some-other-key-name
+        value: some-other-key-value
+```
+
+If you then read back the kdcluster spec from the K8s API, you will see that the "value" property has been removed from each of those list elements, replaced with an "encryptedValue" property. This "encryptedValue" property contains the AES-GCM encrypted form of the original "value" property. It is only for reference by KubeDirector and does not need to be directly interpreted by a user or application. The original "value" property is never visible at any time through the K8s API.
+
+This "secretKeys" list can be specified at kdcluster creation time. When editing a kdcluster, currently you are not allowed to modify this list if the role has existing members.
+
+In situations where editing is allowed, you can freely remove list elements, or add new elements that have "name" and "value" pairs. You are not allowed to modify the "encryptedValue" of an existing list element  -- but you could specify some new "value" instead, which would cause the "encryptedValue" to be updated.
+
+#### ACCESS FROM THE CONTAINER
+
+Within a container, you can use configcli tools to access the original "value" strings, for any secret-keys element specified for the role of this container.
+
+**Note:** As with any use of configcli, this is only possible if the role in the kdapp has a "configPackage" specified.
+
+The values are indexed by name under the "secret_keys" token, which is available for each role ID.
+
+An example of accessing the value for the secret key named "some-key-name", using the ccli utility from the shell:
+```bash
+DISTRO_ID=$(ccli --get node.distro_id)
+NG_ID=$(ccli --get node.nodegroup_id)
+ROLE_ID=$(ccli --get node.role_id)
+SECRET_KEY_NAME=some-key-name
+SECRET_KEY_VALUE=$(ccli --get "distros.$DISTRO_ID.$NG_ID.roles.$ROLE_ID.secret_keys.$SECRET_KEY_NAME")
+```
+
+An example of accessing the value for the secret key named "some-key-name", using the configcli Python module:
+```python
+from configcli import ConfigCli
+configcli = ConfigCli(shell=False)
+namespace = configcli.getCommandObject("namespace")
+distroID = namespace.getWithTokens(["node", "distro_id"])
+ngID = namespace.getWithTokens(["node", "nodegroup_id"])
+roleID = namespace.getWithTokens(["node", "role_id"])
+secretKeyName = "some-key-name"
+secretKeyValue = namespace.getWithTokens(["distros", distroID, ngID, "roles", roleID, "secret_keys", secretKeyName])
```

Comment:
@aleshchynskyi @joel-bluedata  If I remember this correctly, this still gives decrypt value? 

Can we clarify this in the doc.

Reply:
Nope, this will give unencrypted... only the unencrypted value is stored in the configmeta.

Elaborating a bit: this is the value that was originally passed in as the "value" in a secretKeys element. Now it may be the case that this value is ITSELF a decryption key that will be used to decrypt other things (e.g. from a Secret or Configmap), but whatever that usage is, it's application-specific.